sdk/src/client/api/block_builder/input_selection/mod.rs

```diff
@@ -379,7 +380,12 @@ impl InputSelection {
     pub fn select(mut self) -> Result<Selected, Error> {
         if !OUTPUT_COUNT_RANGE.contains(&(self.outputs.len() as u16)) {
             // If burn or mana allotments are provided, outputs will be added later.
-            if !(self.outputs.is_empty() && (self.burn.is_some() || self.mana_allotments != 0)) {
+            if !(self.outputs.is_empty()
+                && (self.burn.is_some()
+                    || self.mana_allotments != 0
+                    // TODO: ??? is this okay ???
+                    || self.available_inputs.iter().any(|o| o.output.is_delegation())))
```

Comment:
The delegation should be added as required_input, right? For the consolidation and claiming PRs (still draft) I changed it to `!self.required_inputs.is_empty()` which I think is better because then you know for sure that these output will be part of the inputs

Reply:
removed

Reply:
Not even `!self.required_inputs.is_empty()` is needed?

Reply:
I think the burn implies that